Make it buildable for wasm with emscripten

HKalbasi opened the pull request:
fix #215

Surprisingly, it just needed to use emscripten's sysroot with bindgen. For `wasm32-wasi` target more work is needed, as it doesn't support c++ exceptions yet, but I think `wasm32-unknown-emscripten` is enough for most of use cases (including my use case).

waywardmonkeys reviewed (changes requested):
How does this build Z3 itself and how does that work?

Could you remove the change to `.gitignore` and do that as a separate thing?

HKalbasi commented:
`cmake` crate is aware that it should use `emcmake` when target is `wasm32-unknown-emscripten` and z3 is ok with emscripten (they use it for js builds). So everything works automatically.

I removed `.gitignore` and added a CI check for wasm build.
